src/main/java/org/fogbowcloud/arrebol/api/http/services/JobService.java

```diff
@@ -28,16 +32,22 @@
 
     private final Logger LOGGER = Logger.getLogger(JobService.class);
 
-    public String addJob(JobSpec jobSpec){
+    public String addJob(JobSpec jobSpec) {
         LOGGER.debug("Creating job object from job specification.");
+        validateJobSpec(jobSpec);
         Job job = createJobFromSpec(jobSpec);
         String id = this.arrebolFacade.addJob(job);
         this.jobDAO.addJob(job);
         return id;
     }
 
-    public Job getJobById(String id){
+    public Job getJobById(String id) {
         Job job = this.jobDAO.getJobById(id);
+        if(job == null){
+            String message = String.format(Messages.Exception.JOB_NOT_FOUND, id);
+            LOGGER.error(message);
```

Comment:
I wonder if we are not logging the same stuff twice (both here and in the point we catch it)

Reply:
Yes, it was repeated in the jobController, I removed the logger that existed in the jobService.